### DISCUSSION TITLE
"ReferenceError: global is not defined" Error in Sveltekit when using @supabase/auth-helpers-sveltekit

### DISCUSSION THREAD

**Discussion by 0xgautam** [supabase/supabase#16804] [Questions]
Hi everyone,

So I am trying to setup Supabase Auth in my Sveltekit application.

I have used the following guide => https://supabase.com/docs/guides/auth/auth-helpers/sveltekit

I have done the exact setup provided here. However, I am getting an error

`ReferenceError: global is not defined - at index.mjs:654:16`

It disappears when I remove the code from  `+layout.ts` file. The error is caused when importing the `createSupabaseLoadClient`.

```
//+layout.ts

import { PUBLIC_SUPABASE_ANON_KEY, PUBLIC_SUPABASE_URL } from '$env/static/public';
import type { Database } from '$lib/supabase.js';
import { createSupabaseLoadClient } from '@supabase/auth-helpers-sveltekit';

export const load = async ({ fetch, data, depends }) => {
	depends('supabase:auth');

	const supabase = createSupabaseLoadClient<Database>({
		supabaseUrl: PUBLIC_SUPABASE_URL,
		supabaseKey: PUBLIC_SUPABASE_ANON_KEY,
		event: { fetch },
		serverSession: data.session
	});

	const {
		data: { session }
	} = await supabase.auth.getSession();

	return { supabase, session };
};
```

I tried looking online but no help. 

Would really appreciate help from the community!

**Comment by GaryAustin1** [comment]
There are several issues on a global error in Supabase discord.  https://discord.com/channels/839993398554656828/1143820546660905001
One answer says: _reverting @supabase/supabase-js back to 2.32.0 seems to prevent the issue_
But there were different environments mentioned so not sure if related to Sveltekit.
